src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/Validator.cs

```diff
@@ -645,5 +702,10 @@ private sealed class ValidationError
 
             internal void ThrowValidationException() => throw new ValidationException(ValidationResult, _validationAttribute, _value);
         }
+
+        private sealed record Path(string Name, int Index = -1)
+        {
+            public override string ToString() => Index == -1 ? Name : $"{Name}[{Index}]";
+        }
```

Comment:
No, you can't put ref structs into collections.

Reply:
(but this also needn't be a record... you can just use a tuple)

Reply:
It seems as though this PR and feature will shortly be scrapped or replaced with something more modern to handle validation. Otherwise, a good idea. I originally went with a `record` as there was some behaviour in it, but that behaviour has since been removed.